Documentation/RepoApi.md

```diff
@@ -28,25 +28,23 @@ There will be a number of actions source-build will need the repos to implement.
 
 1. `Build` - Assemble the product from the current source contained in the repo.
 
-source-build formats arguments using a style that is compatible with MSBuild. The reasoning for this is that the majority of repos have an MSBuild based build system. Using this format will allow for arguments to easily be passed through to MSBuild. Similarly, both action and parameter arguments are case-insensitive, just like MSBuild default behavior.
-
-Actions will take the form of `-t:ActionName`, just like MSBuild targets are passed to `msbuild.exe`.
+Actions will take the form of `ActionName` and are the first parameter passed to the build script. Action names are case insensitive.
 
 ### Parameters
 
 Each action above will take a set of parameters used to pass information into the action and control its behavior.
 
-Parameters will take the form of `-p:ParameterName=ParameterValue`, just like MSBuild properties are passed to `msbuild.exe`. One slight variation here is the usage of `-p` instead of `/p`.  MSBuild supports both forms, but source-build chooses `-p` because it meets [POSIX recommendations](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html) for command line parameters and avoids any confusion with file paths on Unix machines.
+Parameters will take the form of `-ParameterName ParameterValue`. This form avoids any confusion with file paths on Unix machines by not using `/` to start the argument and works well in both powershell and shell script parsing. Every parameter passed by source-build will have a value, even if that value is just `true`.
 
-In order to avoid conflicts between existing MSBuild properties used by the repos and the properties passed in by source-build, source-build will append a prefix to every property it passes in: `DotNet`.
+In order to avoid conflicts between existing properties (MSBuild or otherwise) used by the repos and the properties passed in by source-build, source-build will append a prefix to every property name it passes in: `DotNet`.
 
 #### Detecting build from source
 
 Some repos build more projects than are necessary to build the .NET Core stack. For example, the `dotnet/roslyn` repo builds the C# compiler, which is needed, but it also builds Visual Studio extensions, which aren't needed. These VS extensions cannot be built without using an external dependency on tooling that isn't available when building from source.
 
 In order to tell repos that these projects shouldn't be built, source-build will pass the following parameter
 
-`-p:DotNetBuildFromSource=true`
+`-DotNetBuildFromSource true`
```

Comment:
Could this just be a switch without needing to also pass in `true`?

Reply:
See the above line

> Every parameter passed by source-build will have a value, even if that value is just `true`.

My thinking is that will make the parsing easier - if there is a parameter name, then there is a parameter value.

The way POSIX solves this is to use `=` in between the parameter name and value.  If there is no `=`, then it is a switch.  

Also, when passing through to MSBuild, the parsing code won't need to figure out "which ones are switches?" and use `true` for their values going into MSBuild parameters.

This was my reasoning at least.  Thoughts?

Reply:
>The way POSIX solves this is to use = in between the parameter name and value. If there is no =, then it is a switch.

Is there any harm with using that approach here? Or is that still and issue with PS parsing? I'm sort of hoping the PS issue is only because of the `/p` but perhaps not. 

Even if we don't have switches as part of the repo APIs I do think the arg parsing is likely going to need to handle them because a lot of repo's already support them, so I think the arg parsing complexity is already going to exist.

Reply:
The `=` itself causes issues in the cmd file before passing through to the ps1:
https://github.com/dotnet/source-build/issues/194#issuecomment-333579905
(Complicated workaround:) https://github.com/dotnet/source-build/issues/194#issuecomment-333747630

Reply:
> My thinking is that will make the parsing easier - if there is a parameter name, then there is a parameter value.

I don't see any reason why that would be a problem. I was just wondering